### ISSUE TITLE
Undefined variable 'placeholder' after pipewire cleanup

### ISSUE THREAD

**Issue by rebeka-catalina** [NixOS/nixpkgs#38285]
## Issue description

Can't run software updates because

``
error: undefined variable ‘placeholder’ at /nix/store/f6brq23x3l16612vkhw8wklaz2zvzdyn-nixpkgs-18.09pre133932.ee28e35ba37/nixpkgs/pkgs/development/libraries/pipewire/default.nix:37:46
``

wich came in with commit ``0bf6fb76f5ec8e218b5d93bbb3ef50ee22cb0827``

### Steps to reproduce

For me commands like ``nix-env -u '*'`` or ``nix-env -qaP`` don't work anymore and I don't know what to do now.

## Technical details

- system: `"x86_64-linux"`
 - host os: `Linux 4.9.76-gentoo-r1, Gentoo, noversion`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(rebeka): `"nixpkgs-18.03pre126798.42b9b8f7c86"`
 - channels(root): `"nixpkgs-18.09pre133932.ee28e35ba37"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

**Comment by bjornfor** [comment]
This should be fixed (reverted) now. Try updating the channel.

**Comment by rebeka-catalina** [comment]
On 2018.04.01, Bjørn Forsman wrote:
> This should be fixed (reverted) now. Try updating the channel.

Sorry, no. I still get the same error.
I'm on nixpkgs-unstable - is this relevant in this case?

**Comment by bjornfor** [comment]
Ah, the nixpkgs-unstable channel hasn't updated in a while: https://howoldis.herokuapp.com/ (currently 4 days old). Your options are to wait or to build from nixpkgs master. (Or see if there is anything blocking the channel update and work on that.)

**Comment by LnL7** [comment]
Only the user channel is nixpkgs-unstable, right? See https://gist.github.com/grahamc/c60578c6e6928043d29a427361634df6#which-channel-is-right-for-me

**Comment by bjornfor** [comment]
I don't know what "user" channel is, and I couldn't find out via that URL either.

**Comment by rebeka-catalina** [comment]
On 2018.04.01, Daiderd Jordan wrote:
> Only the user channel is nixpkgs-unstable, right? See https://gist.github.com/grahamc/c60578c6e6928043d29a427361634df6#which-channel-is-right-for-me

I once created a seperate channel as user, but this happened
accidentally so then I deleted the symlink from
~/.nix-defexpr/ to only use one channel.

``
# nix-channel --list
nixpkgs https://nixos.org/channels/nixpkgs-unstable
``

**Comment by LnL7** [comment]
If you look at the nix-info output in the description you can see there are 2 channels on the system.

**Comment by rebeka-catalina** [comment]
On 2018.04.01, Bjørn Forsman wrote:
> I don't know what "user" channel is, and I couldn't find out via that URL either.

As regular user I once did a `nix-channel --update` and then there
were two symlinks in ~/.nix-defexpr. channels_root@ and another one -
I deleted the other one and kept channels_root@

**Comment by rebeka-catalina** [comment]
On 2018.04.01, Daiderd Jordan wrote:
> If you look at the nix-info output in the description you can see there are 2 channels on the system.

I know and I wondered about it, because if I do a nix-channel --list
as root it shows only one - if I do it as user, it also shows only on
and in both scenarios with the same result - so both show the
unstable.

**Comment by LnL7** [comment]
Uh oh, ignore my comment about nixpkgs-unstable. That is only relevant for nixos.

I don't know exactly how `nix-env -u` behaves but it's possible you have to update both channels to avoid the error. Note the ee28e35ba37 (root channel) in the path of the error.

**Comment by rebeka-catalina** [comment]
> I don't know exactly how `nix-env -u` behaves but it's possible you have to update both channels to avoid the error.

I tried it, but now I have again two symlinks :
~/.nix-defexpr/channels@
and
~/.nix-defexpr/channels_root@

and I get a warning:

``
warning: name collision in input Nix expressions, skipping
‘/home/rebeka/.nix-defexpr/channels_root/nixpkgs’
``

That's why I deleted the channels symlink that day. I tried also to
remove the channel for my user, but this doesn't work.
I think I don't need two channels.

But nix-channel --update as user also didn't help.

> Note the ee28e35ba37 (root channel) in the path of the error.

What do mean with this?
